test/HypLSP8.t.sol

```diff
@@ -0,0 +1,218 @@
+        remoteToken.enrollRemoteRouter(DESTINATION, address(remoteToken).addressToBytes32());
+
+        _performRemoteTransfer(25_000, TOKEN_ID);
+        assertEq(lsp8Token.balanceOf(ALICE), 0);
+    }
+
+    function testRemoteTransfer_revert_unauthorizedOperator() public {
+        vm.prank(OWNER);
+        vm.expectRevert("!owner");
+        localToken.transferRemote{ value: 25_000 }(DESTINATION, BOB.addressToBytes32(), uint256(TOKEN_ID));
+    }
+
+    function testRemoteTransfer_revert_invalidTokenId() public {
+        bytes32 invalidTokenId = bytes32(uint256(999));
+        vm.expectRevert(abi.encodeWithSignature("LSP8NonExistentTokenId(bytes32)", invalidTokenId));
+        _performRemoteTransfer(25_000, invalidTokenId);
+    }
+}
+
+contract HypLSP8CollateralTest is HypTokenTest {
+    using TypeCasts for address;
+
+    HypLSP8Collateral internal lsp8Collateral;
+
+    function setUp() public override {
+        super.setUp();
+
+        localToken = new HypLSP8Collateral(address(localPrimaryToken), address(localMailbox));
+        lsp8Collateral = HypLSP8Collateral(address(localToken));
+
+        vm.prank(OWNER);
+        lsp8Collateral.initialize(address(noopHook), address(0), OWNER);
+
+        // Give accounts some ETH for gas
+        vm.deal(OWNER, 1 ether);
+        vm.deal(ALICE, 1 ether);
+        vm.deal(BOB, 1 ether);
+
+        // Mint test tokens
+        vm.startPrank(OWNER);
+        localPrimaryToken.mint(OWNER, TOKEN_ID, true, "");
+        localPrimaryToken.transfer(OWNER, ALICE, TOKEN_ID, true, "");
+        vm.stopPrank();
+
+        // Deploy remote token
+        remoteToken = new HypLSP8(address(remoteMailbox));
+        vm.prank(OWNER);
+        remoteToken.initialize(0, NAME, SYMBOL);
+
+        // Enroll routers for both chains
+        vm.prank(OWNER);
+        lsp8Collateral.enrollRemoteRouter(DESTINATION, address(remoteToken).addressToBytes32());
+
+        vm.prank(OWNER);
+        remoteToken.enrollRemoteRouter(ORIGIN, address(localToken).addressToBytes32());
+    }
+
+    function testRemoteTransfer() public {
+        vm.prank(ALICE);
+        localPrimaryToken.authorizeOperator(address(lsp8Collateral), TOKEN_ID, "");
+        _performRemoteTransfer(25_000, TOKEN_ID);
+
+        assertEq(localPrimaryToken.tokenOwnerOf(TOKEN_ID), address(lsp8Collateral));
+    }
+
+    function testRemoteTransfer_revert_unauthorized() public {
+        vm.expectRevert(
+            abi.encodeWithSignature("LSP8NotTokenOperator(bytes32,address)", TOKEN_ID, address(lsp8Collateral))
+        );
+        vm.prank(BOB);
+        localToken.transferRemote{ value: 25_000 }(DESTINATION, BOB.addressToBytes32(), uint256(TOKEN_ID));
+    }
+
+    function testRemoteTransfer_revert_invalidTokenId() public {
+        bytes32 invalidTokenId = bytes32(uint256(999));
+        vm.expectRevert(abi.encodeWithSignature("LSP8NonExistentTokenId(bytes32)", invalidTokenId));
+        _performRemoteTransfer(25_000, invalidTokenId);
```

Comment:
Why in these tests is it needed to send `25_000` native tokens as msgValue?

<img width="751" alt="image" src="https://github.com/user-attachments/assets/c3c772c9-6fe2-4a6b-bd00-ccb29ac614df">

Reply:
I tihnk the 25_000 native tokens (in wei) are required as msg.value because Hyperlane uses this payment to cover the gas fees needed for cross-chain message processing on the destination chain, ensuring the token transfer can be executed on the remote chain.

Reply:
Ok added in the list of questions on ClickUP. Marking as resolved.